nautobot/extras/jobs.py

```diff
@@ -460,12 +465,29 @@ def as_form(self, data=None, files=None, initial=None, approval_view=False):
 
         form = self.as_form_class()(data, files, initial=initial)
 
-        job_model = JobModel.objects.get_for_class_path(self.class_path)
-        task_queues = job_model.task_queues if job_model.task_queues_override else self.task_queues
+        try:
+            job_model = JobModel.objects.get_for_class_path(self.class_path)
+            read_only = job_model.read_only if job_model.read_only_override else self.read_only
+            dryrun_default = job_model.dryrun_default if job_model.dryrun_default_override else self.dryrun_default
+            task_queues = job_model.task_queues if job_model.task_queues_override else self.task_queues
+        except JobModel.DoesNotExist:
```

Comment:
This had a todo note to remove in 2.0 but it still generates a 500 error when you try to navigate to a scheduled job for a job model that has been deleted. Should this just raise a better error message instead of falling back to the job class?

Reply:
Should we set the FK to delete cascade?